ssl/quic/quic_channel.c

```diff
@@ -2215,6 +2220,76 @@ static void ch_rx_handle_packet(QUIC_CHANNEL *ch, int channel_only)
          */
         return;
 
```

Comment:
We should actually be ok here, because the function ossl_quic_pkt_type_has_version only returns true if the packet type is neither QUIC_PKT_TYPE_1RTT nor QUIC_PKT_TYPE_VERSION_NEG.  So when establishing a connection the first packet we receive from the server should be of type QUIC_PKT_TYPE_VERSION_NEG, meaning we will continue to process it, at which point we fall into the added clause below this check, and restart the connection with the proper version.